modules/external-api/src/main/java/org/opencastproject/external/endpoint/CaptureAgentsEndpoint.java

```diff
@@ -0,0 +1,157 @@
+import static org.opencastproject.util.doc.rest.RestParameter.Type.STRING;
+
+import org.opencastproject.capture.admin.api.Agent;
+import org.opencastproject.capture.admin.api.CaptureAgentStateService;
+import org.opencastproject.external.common.ApiMediaType;
+import org.opencastproject.external.common.ApiResponses;
+import org.opencastproject.external.common.ApiVersion;
+import org.opencastproject.util.doc.rest.RestParameter;
+import org.opencastproject.util.doc.rest.RestParameter.Type;
+import org.opencastproject.util.doc.rest.RestQuery;
+import org.opencastproject.util.doc.rest.RestResponse;
+import org.opencastproject.util.doc.rest.RestService;
+
+import com.entwinemedia.fn.data.json.JValue;
+
+import org.osgi.service.component.ComponentContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+
+@Path("/")
+@Produces({ "application/json", "application/v1.1.0+json" })
+@RestService(
+    name = "externalapicaptureagents",
+    title = "External API Capture Agents Service",
+    notes = "",
+    abstractText = "Provides resources and operations related to the capture agents"
+)
+public class CaptureAgentsEndpoint {
+
+  /** The logging facility */
+  private static final Logger logger = LoggerFactory.getLogger(CaptureAgentsEndpoint.class);
+
+  /** The capture agent service */
+  private CaptureAgentStateService agentStateService;
+
+  /** OSGi DI */
+  public CaptureAgentStateService getAgentStateService() {
+    return agentStateService;
+  }
+
+  /** OSGi DI */
+  public void setAgentStateService(CaptureAgentStateService agentStateService) {
+    this.agentStateService = agentStateService;
+  }
+
+
+  /** OSGi activation method */
+  void activate(ComponentContext cc) {
+    logger.info("Activated External API - Capture Agents Endpoint");
+  }
+
+  @GET
+  @Path("{agentId}")
+  @RestQuery(
+      name = "getagent",
+      description = "Returns a single capture agent.",
+      returnDescription = "",
+      pathParameters = {
+          @RestParameter(name = "agentId", description = "The agent id", isRequired = true, type = STRING)
+      },
+      reponses = {
+          @RestResponse(description = "The agent is returned.", responseCode = HttpServletResponse.SC_OK),
+          @RestResponse(description = "The specified agent does not exist.", responseCode = HttpServletResponse.SC_NOT_FOUND)
+      }
+  )
+  public Response getAgent(
+      @HeaderParam("Accept") String acceptHeader,
+      @PathParam("agentId") String id) throws Exception {
+    final ApiVersion requestedVersion = ApiMediaType.parse(acceptHeader).getResponseVersion();
```

Comment:
As this line appear so often, it might be worth thinking about overloading ApiResponses.Json.ok to also directly support the accetHeader as formal parameter

Reply:
ok, done